gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/resources/liquibase/changelogs/v4_4_0/05_update_category_keys_to_ids_in_apis_table.yml

```diff
@@ -0,0 +1,22 @@
+databaseChangeLog:
+  - changeSet:
+      id: 4.4.0_update_category_keys_to_ids_in_apis_table
+      author: GraviteeSource Team
+      changes:
+        - sql:
+            sql: |
+              -- Create a temporary table to store the mapping from category keys to IDs
+              CREATE TEMPORARY TABLE temp_category_mapping AS
+              SELECT _id, key
+              FROM categories;
+
+              -- Update the categories column in the apis table
+              UPDATE apis
+              SET categories = (
+                SELECT array_agg(COALESCE(temp_category_mapping._id, apis.categories[i]))
+                FROM generate_subscripts(apis.categories, 1) AS s(i)
```

Comment:
Are you sure this will work with MySQL, MariaDB and SQLServer ?

Reply:
@phiz71 I've decided to add batching for both Mongo and JDBC